src/lib/logical_query_plan/static_table_node.cpp

```diff
@@ -100,8 +107,36 @@ std::shared_ptr<AbstractLQPNode> StaticTableNode::_on_shallow_copy(LQPNodeMappin
 
 bool StaticTableNode::_on_shallow_equals(const AbstractLQPNode& rhs, const LQPNodeMapping& /*node_mapping*/) const {
   const auto& static_table_node = static_cast<const StaticTableNode&>(rhs);
-  return table->column_definitions() == static_table_node.table->column_definitions() &&
-         table->soft_key_constraints() == static_table_node.table->soft_key_constraints();
+
+  if (table->column_definitions() != static_table_node.table->column_definitions()) {
+    return false;
+  }
+
+  // If the column definitions match, we check if the valid key constraints are also equal then we assume equality.
+  const auto& key_constraints = table->soft_key_constraints();
+  const auto& rhs_key_constraints = static_table_node.table->soft_key_constraints();
```

Comment:
Can we use just `==` for the key constraints? `StaticTableNode`s are in essence used for meta tables and for create table statements. In both cases, there sould not be invalidated constraints.